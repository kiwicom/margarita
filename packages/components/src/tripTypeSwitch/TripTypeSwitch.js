// @flow

import * as React from 'react';
import { View } from 'react-native';
import { StyleSheet, type StylePropType } from '@kiwicom/universal-components';

import TripTypeSwitchOption from './TripTypeSwitchOption';

type Props = {|
  +style?: StylePropType,
  +optionsData: {
    [string]: {|
      +icon: string,
      +label: string,
    |},
  },
  +selectedType: string,
  +onSelect: (type: string) => void,
|};

export default function TripTypeSwitch({
  style,
  optionsData,
  selectedType,
  onSelect,
}: Props) {
  const optionKeys = Object.keys(optionsData);
  const options = optionKeys.map(key => {
    const optionData = optionsData[key];
    return (
      <TripTypeSwitchOption
        key={key}
        label={optionData.label}
        type={key}
        selected={selectedType === key}
        onSelect={onSelect}
      />
    );
  });
  return <View style={[styles.container, style]}>{options}</View>;
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'flex-end',
    marginEnd: 20,
  },
});
