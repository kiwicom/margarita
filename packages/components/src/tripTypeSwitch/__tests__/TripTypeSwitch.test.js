// @flow

import * as React from 'react';
import { shallow, render, fireEvent } from 'react-native-testing-library';

import TripTypeSwitch from '../TripTypeSwitch';

const optionsData = {
  Return: {
    icon: 'flight-return',
    label: 'Return',
  },
  OneWay: {
    icon: 'flight-direct',
    label: 'One way',
  },
};
const onSelect = jest.fn();

const { getByText } = render(
  <TripTypeSwitch
    optionsData={optionsData}
    selectedType={'Return'}
    onSelect={onSelect}
  />,
);

it('should execute onSelect method with correct value', () => {
  const testKey = Object.keys(optionsData)[1];
  fireEvent(getByText(optionsData[testKey].label.toUpperCase()), 'press');
  expect(onSelect).toHaveBeenCalledWith(testKey);
});

it('renders', () => {
  expect(
    shallow(
      <TripTypeSwitch
        optionsData={optionsData}
        selectedType={'Return'}
        onSelect={jest.fn()}
      />,
    ),
  ).toMatchSnapshot();
});
