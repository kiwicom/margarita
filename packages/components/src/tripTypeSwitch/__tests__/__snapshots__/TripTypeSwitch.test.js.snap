// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "output": <Component
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "marginEnd": 20,
        },
        undefined,
      ]
    }
  >
    <TripTypeSwitchOption
      label="Return"
      onSelect={[MockFunction]}
      selected={true}
      type="Return"
    />
    <TripTypeSwitchOption
      label="One way"
      onSelect={[MockFunction]}
      selected={false}
      type="OneWay"
    />
  </Component>,
}
`;
